### PR TITLE
IRC can namepick

### PR DESCRIPTION
**Pull request by craftxbox**
**What does this PR do:**
Lets robobrains placed into irc chassis namepick when placed into chassis

**Changelog:**
:cl:
add: irc now get namepick prompt when placed in chassis
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
They're in the same category as golems; they're not meant to be able to pick their names; they're non-sentient created slaves to do your bidding; they have no concept of identity or personhood.

👎

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -5,6 +5,7 @@
 	icon = 'icons/mob/human.dmi'
 	icon_state = "body_m_s"
 	deathgasp_on_death = TRUE
+	var/mob/living/carbon/human/cannamepick = FALSE
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/human.dm]: What...are you doing?

File: code/modules/mob/mob_helpers.dm [hunk 2]
```diff
@@ -584,11 +584,19 @@ var/list/intents = list(INTENT_HELP,INTENT_DISARM,INTENT_GRAB,INTENT_HARM)
...
 			if(newname)
 				break	//That's a suitable name!
-			to_chat(src, "Sorry, that [role]-name wasn't appropriate, please try another. It's possibly too long/short, has bad characters or is already taken.")
+			to_chat(src, "Sorry, that [role] name wasn't appropriate, please try another. It's possibly too long/short, has bad characters or is already taken.")
 
 		if(!newname)	//we'll stick with the oldname then
 			return
 
+		if(ishuman(src))
+			var/mob/living/carbon/human/H = src
```
Comment (Fox-McCloud) [inline code/modules/mob/mob_helpers.dm]: This is an egregious violation of OOP. Don't do this.

File: code/modules/mob/living/carbon/human/human.dm (hunk 1 above)
Comment (craftxbox) [reply code/modules/mob/living/carbon/human/human.dm]: making sure that they cant namepick twice, or that other species cant nampick

File: code/modules/mob/mob_helpers.dm (hunk 2 above)
Comment (craftxbox) [reply code/modules/mob/mob_helpers.dm]: had no clue what else to do to cast it.

File: code/modules/mob/living/carbon/human/human.dm (hunk 1 above)
Comment (Fox-McCloud) [reply code/modules/mob/living/carbon/human/human.dm]: Why are you typecasting a variable only for it to be `TRUE` or `FALSE`.

File: code/modules/mob/mob_helpers.dm [hunk 3]
```diff
@@ -584,11 +584,19 @@ var/list/intents = list(INTENT_HELP,INTENT_DISARM,INTENT_GRAB,INTENT_HARM)
...
+			if(H.cannamepick == FALSE)
```
Comment (Fox-McCloud) [inline code/modules/mob/mob_helpers.dm]: `!cannamepick`

File: code/modules/mob/mob_helpers.dm (hunk 2 above)
Comment (Fox-McCloud) [reply code/modules/mob/mob_helpers.dm]: It needs to be more OOP; this should be under the human level proc, not at the mob level.

File: code/modules/mob/mob_helpers.dm (hunk 3 above)
Comment (craftxbox) [reply code/modules/mob/mob_helpers.dm]: Wasnt sure if byond had that operator, so i stood on the safe side.

**Comment by Aurorablade** [comment]
Echoing, They are in the same catagory as golems. So this..yeahhhhhh defeats the point.

**Comment by craftxbox** [comment]
I mean, borgs are also in the same category, but they get namepick

**Comment by Aurorablade** [comment]
Borgs aren't in the same catagory as golems and IRCs at all. Borgs have some free thinking restricted by walls. Golems/IRCs aren't supposed to be freewilled at all.
The issue at this point seems to be the ID computer and if you want to allow numbers, however you can also just make a new manifest entry free named and apply it to a blank ID based on if the person they serve/imprinted on wishes for them to do that.

**Comment by craftxbox** [comment]
But theyre both made from the same source brain, but for some reason putting them in a borg chassis gives them more freedom than an IR Chassis?

**Comment by granodd** [comment]
What if they can name pick if you stick in an MMI/Posibrain as opposed to the enslaved robotic circuits?

**Comment by PidgeyThePirate** [comment]
Still dumb. If you can namepick as a borg, which is, when created with a robotic brain, supposed to be a nonsentient servant, you can and should namepick as an IRC.

**Comment by Dinarzad** [comment]
I don't really see a problem with letting them pick a name. "It defeats the point" is an absolutely silly argument because the "Point" of an IRC is to have a golem-esque servant that you can make.
There's nothing that really says they have no sense of sentience or self, they are simply lawed/slaved to their creator. Similar to how IPCs were once bound by laws in their lore before getting a sort of civil rights movement together an getting freedom.

"IPCs (Integrated Positronic Chassis) were meant to serve as enhanced synthetic assistants. With minds akin to organic beings, their advanced problem-solving abilities rendered them capable of assisting with many complicated tasks related to science and engineering."

"Despite being fully sapient, IPCs were regarded as station equipment in a similar manner to cyborgs - with individual units going unpaid, unserviced, and typically recycled when they sustained critical damage."

**Comment by Superhats** [comment]
IRC aren't supposed to have an identity. It's the one thing that differentiates them from actual IPC. It's also one of the main reasons IRC were added in the first place. IRC not being able to pick a name is fully intended, and it should stay that way. 
Robotic brains are explicitly non-sentient, unlike IPC.

**Comment by PidgeyThePirate** [comment]
Then take out borgs made with robotic brains being able to choose their own names. Either both must be true or none for consistency. Alternatively, you could allow the roboticist to rename an IRC chassis instead so you strip the bot's agency as you wish. You have to make that true for a robotic-brain borg too though.

**Comment by Couls** [comment]
Yeah I personally think it’d make more sense for the roboticist to name their creations if we’re going the no free will route

**Comment by granodd** [comment]
@Couls borgs are still created form the brains of living, sentient people and machines. They still retain their personality and thoughts, though are subjected to a set of laws that they **must** follow.

**Comment by Breenland** [comment]
I don't see any issue with them getting to pick their names.

File: code/modules/mob/mob_helpers.dm (hunk 2 above)
Comment (craftxbox) [reply code/modules/mob/mob_helpers.dm]: is there even a human level proc at the moment?

**Comment by PidgeyThePirate** [comment]
@Granodd No, they're also created from robotic brains which are by nature nonsentient. The lack of sentience isn't a feature of the IRC (which can also be piloted by a sentient MMI or positronic brain), it's a feature of the brain. So either borgs and IRCs can namepick or they can't, it's a package deal.

**Comment by EvadableMoxie** [comment]
I'm fine with IRCs being able to namepick, but if they can then there needs to be some other kind of reliable indicator that you're dealing with an IRC and not an IPC.  

IPC are crew and entitled to full protection under space law. IRCs are station property.  If they're indistinguishable then that's a problem.  If they aren't covered you can probably tell by examining them, but if they're in a hardsuit or something that conceals their species that could be very problematic.

**Comment by rb303** [comment]
I dunno, they're just meant to be robot golems; giving them names just discourages this further and kinda just opens up the option of people becoming IPC at roundstart by running to robotics without needing to pay karma for it.

**Comment by Dinarzad** [comment]
> I dunno, they're just meant to be robot golems; giving them names just discourages this further and kinda just opens up the option of people becoming IPC at roundstart by running to robotics without needing to pay karma for it.

That's literally what you can  right now though...?
Like.
This doesn't change anything all it does is let you pick a name, I am not at all sure why this is such a controversial subject. What is the genuine difference being made in gameplay or lore between being called "Beep boop" or "ARMA" or "110010100101001"
It plays nicer with ID machines to be able to rename yourself and giving the roboticist the ability to name the creation usually ends with them naming things "DICK PUNCHER 9000"

IF we're concerned about giving people IPC without the Karma, then we really should have opposed IRCs in general because they are literally just slaved IPC units. You are a golem in function, but your body is 100% IPC. And I'm not at all sure how the ability to change a name will change that.

> I'm fine with IRCs being able to namepick, but if they can then there needs to be some other kind of reliable indicator that you're dealing with an IRC and not an IPC.

Maybe something that shows up on examination? Like some Shadowling Thrall-esque red text mentioning something about how they don't move quite right/really stiffly, some kind of uncanny valley sort of thing? I'm not sure what other indicator you'd be able to get while they're inside a suit unless you go the Golem route. Not letting them wearing clothes or hardsuits, but giving them the ability to be space worthy or still have IDs. OR just don't let them be space worthy since they're a lot easier to put together then a golem an are made to be mass produced/disposable.

**Comment by Spacemanspark** [comment]
I don't see the problem with letting them pick a name.

**Comment by Superhats** [comment]
We won't be able to differentiate them from IPC

**Comment by Superhats** [comment]
IRC are not IPC

**Comment by Spacemanspark** [comment]
So maybe have their name be 'Integrated Robotic Chassis (`NAME`)' or something.

There's not really an issue with it, considering cyborgs can pick their own names.

Tbh the only difference between an IRC and an IPC is the robotic brain... which we haven't really ever established as being "non sentient". Nobody plays them as "non sentient". 

If that's the case, then hoo boy are admins going to have a lot of work telling people to act like uninteresting personalities now.

**Comment by Superhats** [comment]
Cyborgs get to pick a name because unlike IRC, cyborgs don't have a karma race who looks **Identical**, both visually and mechanically.

If we let IRC namepick freely, robotics will just become a way of turning everyone into IPC, and/or creating a bunch of mindslaves that are impossible to differentiate from IPC.

You forget the extremely important fact that unlike IPC, IRC are **slaved** to their creator. So, a malignant roboticist could very much send an IRC off to the workplace of whichever target they have, have them disguise as a normal IPC because, once again, we will have NO way of telling they are an IRC.

If IRC are able to choose a name, it must be very obvious that they are **not** an IPC.

**Comment by Superhats** [comment]
There's also the fact that IPC are a karma race. Letting IRC namepick freely makes that completely redundant because, why spend 15 karma points on a race when you can just become one at the roboticist?

**Comment by Spacemanspark** [comment]
>You forget the extremely important fact that unlike IPC, IRC are slaved to their creator. So, a malignant roboticist could very much send an IRC off to the workplace of whichever target they have, have them disguise as a normal IPC because, once again, we will have NO way of telling they are an IRC.

As I said before:
"So maybe have their name be 'Integrated Robotic Chassis ( NAME )' or something."

>There's also the fact that IPC are a karma race. Letting IRC namepick freely makes that completely redundant because, why spend 15 karma points on a race when you can just become one at the roboticist?

I don't feel that picking a name changes anything about this at all. If anything, that's more of an argument to remove IRCs. We aren't talking about unslaving them or anything like that, we're talking about a name.

**Comment by EvadableMoxie** [comment]
>why spend 15 karma points on a race when you can just become one at the roboticist?

Many reasons.

There's no guarantee you'll get the brain, or even if you do that the roboticist will give you a choice between IRC or Borg. You can't roll for an antag or the job of your choice at round start. To get a job, you have to hope your master allows it, then go to the HoP and then ask tor the job you want mid-shift and hope the HoP lets you have it.  That's not always reliable and the job you want might not always be available. By the time it does happen you might be pretty deep into the shift. Your master might not be willing to let you just act with autonomy.  They might have their own ideas on what you should be doing, or might be an antag with an agenda. 

So, I don't really think IPCs being a karma race is a good argument against IRCs namepicking, personally. My only issue with it is that I want IRCs to be easily distinguishable at all times.  Spark's idea of adding the namepick to the end would work for me.

**Comment by taukausanake** [comment]
I never thought IRCs were a good addition because of IPCs.

Isn't there some semi-old android sprite out there that's just a robot human? Wouldn't that be a better sprite than just IPC looks?

**Comment by PidgeyThePirate** [comment]
Call me crazy, but I actually think people *should* be able to become karma races over the course of a shift, should they put their back into it. The divide between karma and non-karma leads to some really arbitrary differences and balance concerns which annoy me. So that argument doesn't hold much water in my eyes. I honestly see karma as more of an obstacle and an annoyance than something useful and productive.

Being able to build a self-aware humanoid robot in a retrofuturistic game on a cutting-edge science station surrounded by some of the most advanced tech in the entire galaxy is not something that should be stifled "because raisins".

**Comment by variableundefined** [comment]
Closing for the reasons Fox stated above.

**Comment by Spacemanspark** [comment]
RIP

**Pull request closed without merging**